### ISSUE TITLE
Missing signature for verification of root filesystem

### ISSUE THREAD

**Issue by Spindel** [tianon/docker-brew-debian#46]
Hi,
  We're working on establishing trust for Docker containers, and for that to work, there needs to be a signature on the rootfs.tar.xz file.

Is it possible to add a (detached) signature in the relevant commits?

**Comment by tianon** [comment]
Unfortunately, these rootfs tarballs aren't (and can't currently reasonably be) built on the machine where I keep my GPG key, but I am interested in changing that somehow to make the provenance even more clear here, so I'm going to leave this open as an official "request for enhancement". :+1:

**Comment by Spindel** [comment]
Thank you, I understand the concerns, but want to (re)state our concern here in that it's difficult to track the origins, something that's even worse by Debians design decision to not have a signed package manifest, but to only sign the apt lists.

We aren't trying to rebuild the Docker infrastructure here, but we're trying to take a known container and verify it's origin binaries, something that's turned out to be far harder than anticipated.

**Comment by rvalle** [comment]
I came to this thread also trying to stablish up to which point the official debian docker image can be trusted.

It is certainly impossible to stablish some kind of trust right now. Even considering that the image is being managed by members of the Debian dev team.

This also brings some questions regarding what docker means by "official". 

Alpine linux is now distributing a GPG signed minimal rootfs to be used for containers, just like any other flavor of distribution.

I think debian should do the same. The Dockerfile here could download the rootfs from Debian and verify the GPG signature.

I also wonder weather it is feasible to generate the rootfs from the netinstall.iso, which is signed, in the Dockerfile.

**Comment by tianon** [comment]
This repository is now deprecated in favor of https://github.com/debuerreotype/docker-debian-artifacts, which builds reproducible Debian rootfs tarballs using https://github.com/debuerreotype/debuerreotype. :+1:

To verify the new tarballs, I'd recommend grabbing `debuerreotype` and regenerating them externally.  For all but `wheezy`, the results should be bit-for-bit identical to what I've generated. 
 For `wheezy`, using `diffoscope` on what I've published and the newly generated bits should show a single benign change, which is documented in `debuerreotype`'s README file.
